src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -487,6 +490,85 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - This method is called when the user clicks the "export message history" button
+    //   in the query palette
+    // Arguments:
+    // - text - the text to export
+    // Return Value:
+    // - <none>
+    void TerminalPage::_OnExportChatHistoryRequested(const IInspectable& /*sender*/, const winrt::hstring& text)
+    {
+        time_t nowTime;
+        time(&nowTime);
+
+        tm nowTm;
+        localtime_s(&nowTm, &nowTime);
+
+        wchar_t buf[64];
+        wcsftime(&buf[0], ARRAYSIZE(buf), L"%F %T", &nowTm);
+
+        const auto defaultFileName = RS_(L"TerminalChatHistoryDefaultFileName") + winrt::to_hstring(buf);
+        _SaveFileHelper(text, L"", defaultFileName);
+    }
+
+    fire_and_forget TerminalPage::_SaveFileHelper(const winrt::hstring& text, const winrt::hstring& filepath, const std::wstring_view filename)
+    {
+        // This will be used to set up the file picker "filter", to select .txt
+        // files by default.
+        static constexpr COMDLG_FILTERSPEC supportedFileTypes[] = {
+            { L"Text Files (*.txt)", L"*.txt" },
+            { L"All Files (*.*)", L"*.*" }
+        };
+        // An arbitrary GUID to associate with all instances of this
+        // dialog, so they all re-open in the same path as they were
+        // open before:
+        static constexpr winrt::guid clientGuidExportFile{ 0xF6AF20BB, 0x0800, 0x48E6, { 0xB0, 0x17, 0xA1, 0x4C, 0xD8, 0x73, 0xDD, 0x58 } };
```

Comment:
If you use the same GUID for every Save File operation, it will remember the saving folder regardless of what you are saving.

Put another way: the GUID disambiguates _types of things being saved_, in case the user wants to save all Terminal Buffers to one place and all Chat Histories to another place.